server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -195,6 +222,11 @@ protected void customizeProxyRequest(Request proxyRequest, HttpServletRequest re
   protected URI rewriteURI(HttpServletRequest request)
   {
     final String host = (String) request.getAttribute(HOST_ATTRIBUTE);
+    return rewriteURI(request, host);
+  }
+
+  protected URI rewriteURI(HttpServletRequest request, String host)
```

Comment:
it's possible, but the method I need in URIBuilder setQuery(String) is deprecated, but I guess it'll be better since it will fix IPv6 hosts.